app/code/Magento/CatalogUrlRewrite/Test/Unit/Model/Storage/DynamicStorageTest.php

```diff
@@ -0,0 +1,327 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CatalogUrlRewrite\Test\Unit\Model\Storage;
+
+use Magento\Catalog\Model\ResourceModel\Product;
+use Magento\Catalog\Model\ResourceModel\ProductFactory;
+use Magento\CatalogUrlRewrite\Model\CategoryUrlPathGenerator;
+use Magento\CatalogUrlRewrite\Model\Storage\DynamicStorage;
+use Magento\Framework\Api\DataObjectHelper;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Select;
+use Magento\UrlRewrite\Model\OptionProvider;
+use Magento\Store\Model\ScopeInterface;
+use Magento\UrlRewrite\Service\V1\Data\UrlRewriteFactory;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use ReflectionMethod;
+
+class DynamicStorageTest extends TestCase
+{
+    /**
+     * @var DynamicStorage
+     */
+    private $object;
+
+    /**
+     * @var UrlRewriteFactory|MockObject
+     */
+    private $urlRewriteFactory;
```

Comment:
Please add "Mock" suffix to all mocks
```suggestion
    private $urlRewriteFactoryMock;
```